app.js

```diff
@@ -138,19 +139,20 @@ function initApp(options) {
  * @param {Application} app the application object to load routes into
  * @return {bluebird} a promise resolving to the app object
  */
-function loadRoutes(app) {
+function loadRoutes(app, dir) {
 
-    // get the list of files in routes/
-    return fs.readdirAsync(`${__dirname}/routes`).map((fname) => {
+    // recursively load routes from .js files under routes/
+    return fs.readdirAsync(dir).map((fname) => {
         return BBPromise.try(() => {
-            // ... and then load each route
-            // but only if it's a js file
-            if (!/\.js$/.test(fname)) {
-                return undefined;
+            const resolvedPath = path.resolve(dir, fname);
+            const isDirectory = fs.statSync(resolvedPath).isDirectory();
+            if (isDirectory) {
+                loadRoutes(app, resolvedPath);
```

Comment:
Why not returning the promise? This will start loading several directories is parallel why we better avoid in case some of the routes files do some expensive initialization.

Reply:
I see your point but I'm not quite sure what you have in mind as for what this should look like instead.  Maybe it would be better to recurse over the directory and get a set of promises to load the route files, then feed the set to BBPromise.all()?

Reply:
nono, just add `return` before and we're good

Reply:
Maybe I'm missing something, but the only way I can make this work with a simple 

  if (isDirectory) {
    return loadRoutes(app, resolvedPath);
  }

is if I add another conditional after the check for undefined routes below to keep from triggering the TypeError:

  if (route === undefined) {
    return undefined;
  }
  if (route instanceof Function) {
    return BBPromise.try(() => route);
  }
  // check that the route exports the object we need
  [...]

Reply:
Makes sence. Given that we've shifted our opinions towards concurrent loading I think this is fine.